File: Source/Lib/RCCService/Types/ThumbnailCommand.ts
import { GridCommand } from './GridCommand';
import { ThumbnailSettings } from './ThumbnailSettings';

export class ThumbnailCommand extends GridCommand {
	public Mode: string = 'Thumbnail';
	public MessageVersion: number = 1;
	public Settings: ThumbnailSettings;

	public constructor(settings: ThumbnailSettings) {
		super();
		this.Settings = settings;
	}
}

File: lib/src/grid/commands/Commands/ThumbnailCommand.cs
namespace Grid.Commands;

/// <summary>
/// Command for rendering thumbnails.
/// </summary>
public class ThumbnailCommand : GridCommand
{
    /// <inheritdoc cref="GridCommand.Mode"/>
    public override string Mode { get; }

    /// <inheritdoc cref="GridCommand.MessageVersion"/>
    public override int MessageVersion => 1;

    /// <summary>
    /// The settings for the command.
    /// </summary>
    public ThumbnailSettings Settings { get; }

    /// <summary>
    /// Construct a new instance of <see cref="ThumbnailCommand"/>
    /// </summary>
    /// <param name="settings">The settings for the command.</param>
    public ThumbnailCommand(ThumbnailSettings settings)
    {
        Settings = settings;

        Mode = Settings.Type == ThumbnailCommandType.TexturePack 
            ? "ExecuteScript" 
            : "Thumbnail";
    }
}
